### DISCUSSION TITLE
Lisp & Basic in Cromix?

### DISCUSSION THREAD

**Discussion by KedalionDaimon** [udo-munk/z80pack#264] [Q&A] [answered]
First, thank you helping me with the C compiler - when I now downloaded the disk from your GitHub directly, suddenly it all was where it should be.

Secondly, you have there a BASIC and a LISP in tgz files - how does one attach and start those? Thank you in advance!

**Comment by udo-munk** [comment, marked as answer]
BASIC and LISP are CDOS versions, not Cromix, don't know if they work, never tried.
Cromix can read CDOS disks, put the disk images into an empty drive and use cdoscopy under Cromix to copy files over. Maybe use z80pack 1.38-dev, it has support for running Cromix from a harddisk, so there is a lot of free space available to import more software into the system.

**Reply by udo-munk** [reply]
Oh, and the .tgz files are gzipped tar archives, you need to untar them of course to get the disk images inside.

**Reply by KedalionDaimon** [reply]
Thank you very kindly for pointing me in the right direction. I understand they won't run - when I try lisp, through lisp.com, it DOES actually start up, but then the entire emulator crashes with this on the console:

E (300210097) system: Op-code trap at f864 ed e5

**Reply by KedalionDaimon** [reply]
I have to correct myself! Lisp crashes - but SBASIC actually does work!

# sbasic


CROMEMCO 32K STRUCTURED BASIC version 03.65
Copyright (c) 1977, 1979 Cromemco, Inc.          


>>PRINT 12
12

>>10 FOR I=1 TO 10
>>20 PRINT I\*I\*I 
>>30 NEXT I
>>run
1
8
27
64
125
216
343
512
729
1000

To have that, as you supposed, the diskspace was not quite enough... so I nuked the commands "version" and "tarz80", finding both quite inessential.

**Comment by udo-munk** [comment]
This is why I suggested to use release 1.38-dev with harddisk support. You do this with "git checkout dev" and then rebuild the machines. A 10MB harddisk image is included, it was installed from the cromix_work.dsk and has the same contents. And you have plenty free space and don't need to remove anything.

There also is a CP/M 2.2 emulator which might help to run CP/M programs because Cromix of course is CDOS compatible to some degree. Manual available here:
 https://www.autometer.de/unix4fun/z80pack/ftp/cromemco/Wheatcroft%20Hards%20SIM22%20CPM%20Simulator.pdf
I doubt it helps runnung Lisp though, because it was written for CDOS, on which it runs OK of course.

**Reply by KedalionDaimon** [reply]
Tried the dev branch and confirm it, as far as observeable, as working with the hdd (checked with "free") on Ubuntu 22.04. Actually, even without sim22, a couple of CP/M programs DO seem to work just nicely, in particular also Migues Vis' TE text editor (variant TE_TAK) - if you want to try it: https://github.com/MiguelVis/te ... also lots of the stuff from retroarchive, including the Lisp over there! :)
